test/e2e/serial/workload_overhead_test.go

```diff
@@ -0,0 +1,249 @@
+
+				By(fmt.Sprintf("Scheduling the testing deployment with RuntimeClass=%q", rtClass.Name))
+				var deploymentName string = "test-dp"
+				var replicas int32 = 1
+
+				podLabels := map[string]string{
+					"test": "test-dp",
+				}
+				nodeSelector := map[string]string{}
+				deployment := objects.NewTestDeployment(replicas, podLabels, nodeSelector, fxt.Namespace.Name, deploymentName, objects.PauseImage, []string{objects.PauseCommand}, []string{})
+				deployment.Spec.Template.Spec.SchedulerName = schedulerName
+				deployment.Spec.Template.Spec.Containers[0].Resources.Limits = podResources
+				deployment.Spec.Template.Spec.RuntimeClassName = &rtClass.Name
+
+				err = fxt.Client.Create(context.TODO(), deployment)
+				Expect(err).NotTo(HaveOccurred(), "unable to create deployment %q", deployment.Name)
+
+				By("waiting for deployment to be up&running")
+				dpRunningTimeout := 1 * time.Minute
+				dpRunningPollInterval := 10 * time.Second
+				err = e2ewait.ForDeploymentComplete(fxt.Client, deployment, dpRunningPollInterval, dpRunningTimeout)
+				Expect(err).NotTo(HaveOccurred(), "Deployment %q not up&running after %v", deployment.Name, dpRunningTimeout)
+
+				nrtListPostCreate, err := e2enrt.GetUpdated(fxt.Client, nrtListInitial, 1*time.Minute)
+				Expect(err).ToNot(HaveOccurred())
+
+				By(fmt.Sprintf("checking deployment pods have been scheduled with the topology aware scheduler %q and in the proper node %q", schedulerName, targetNodeName))
+				pods, err := schedutils.ListPodsByDeployment(fxt.Client, *deployment)
+				Expect(err).NotTo(HaveOccurred(), "Unable to get pods from Deployment %q:  %v", deployment.Name, err)
+
+				podResourcesWithOverhead := podResources.DeepCopy()
+				podResourcesWithOverhead.Cpu().Add(*rtClass.Overhead.PodFixed.Cpu())
+				podResourcesWithOverhead.Memory().Add(*rtClass.Overhead.PodFixed.Memory())
+
+				for _, pod := range pods {
+					Expect(pod.Spec.NodeName).To(Equal(targetNodeName))
+					schedOK, err := nrosched.CheckPODWasScheduledWith(fxt.K8sClient, pod.Namespace, pod.Name, schedulerName)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(schedOK).To(BeTrue(), "pod %s/%s not scheduled with expected scheduler %s", pod.Namespace, pod.Name, schedulerName)
+
+					By(fmt.Sprintf("checking the resources are accounted as expected on %q", pod.Spec.NodeName))
+					nrtInitial, err := e2enrt.FindFromList(nrtListInitial.Items, pod.Spec.NodeName)
+					Expect(err).ToNot(HaveOccurred())
+					nrtPostCreate, err := e2enrt.FindFromList(nrtListPostCreate.Items, pod.Spec.NodeName)
+					Expect(err).ToNot(HaveOccurred())
+
+					_, err = e2enrt.CheckZoneConsumedResourcesAtLeast(*nrtInitial, *nrtPostCreate, podResources)
+					Expect(err).ToNot(HaveOccurred())
+
+					// Here error have to Ocurr because pod overhead resources should not be count
+					_, err = e2enrt.CheckZoneConsumedResourcesAtLeast(*nrtInitial, *nrtPostCreate, podResourcesWithOverhead)
+					Expect(err).To(HaveOccurred())
```

Comment:
The test looks good. 
some checks are missing though as per the description in the TP: 
- Total pod resources ( + overhead) should be reflected under the node description. (I am not sure if there is an API to this) 
- there should be an additional step: modify the pod overhead in such a way that the total (pod  +overhead) resources don't fit on any node, thus the pod should stay pending.

Reply:
Jira ticket to track this CNF-4572

Reply:
the `node.Status.Allocatable` is NOT updated when the resource are consumed (that's a common gotcha) so the first point needs to be postponed/ removed

the second point is fair, will add the additional step in a followup PR, I'd prefer to keep this one as simple as possible (ideally code movement + trivial fixes, no new logic if we can help it)

Reply:
thanks for your reply.
point 1: I meant that the resources consumption per pod should be updated under the node description under the `Non-terminated Pods` list, those will include both pod +overhead resources:

```
Non-terminated Pods:                      (50 in total)
  Namespace                               Name                                                 CPU Requests  CPU Limits  Memory Requests  Memory Limits  Age
  ---------                               ----                                                 ------------  ----------  ---------------  -------------  ---
  default                                 helper-deploymentoops-again-5869455848-2ctbg         3 (2%)        3 (2%)      3G (3%)          3G (3%)        4d3h
  default                                 helper-deploymentoops-again-5869455848-8dfhw         3 (2%)        3 (2%)      3G (3%)          3G (3%)        4d3h
  default                                 helper-deploymentoops-again-5869455848-rvxz9         3 (2%)        3 (2%)      3G (3%)          3G (3%)        4d3h

```



I agree on point 2.

Reply:
I need to look deeper into that to assess if/how this is feasible (that's a good point I didn't consider before!) so I'll need to defer to a new PR same as for point 2

Reply:
ok np, I believe this is expected behavior regardless of the scheduler or any other NRO component. Plus, this is the only guarantee that the overhead was actually applied to the pod & yet the pod was scheduled as anticipated.